### PR TITLE
Auth docker compose

### PR DESCRIPTION
**Pull request by jaylees14**
When outputting an auth service for docker compose we need to use a special script that waits for Kong to be available before starting the service. See https://github.com/TempleEight/spec-golang/pull/20 for a full justification.

This adds support for that, as well as some unit tests.

### CODE REVIEW

**Review by Lewiky — APPROVED** [review]
LGTM my dude

File: src/main/scala/temple/builder/DockerfileBuilder.scala [hunk 1]
```diff
@@ -8,10 +9,29 @@ import temple.generate.docker.ast.{DockerfileRoot, Statement}
...
 
+    // If using DockerCompose and this is the auth service, change the exec
+    val execStatements: Option[Seq[Statement]] = service match {
+      case AbstractServiceBlock.AuthServiceBlock =>
+        provider.collect {
+          case Provider.DockerCompose =>
+            Seq(
+              Run("apk", Seq("add", "curl")),
+              Run("chmod", Seq("+x", "wait-for-kong.sh")),
+              Cmd("./wait-for-kong.sh", Seq("kong:8001", "--", "./auth")),
```
Comment (willjvsmith) [inline src/main/scala/temple/builder/DockerfileBuilder.scala:29]: Is Kong always on 8001, or did we pick that

**Review by willjvsmith — APPROVED** [review]
LGTM 🚀

File: src/main/scala/temple/builder/DockerfileBuilder.scala (hunk 1 above)
Comment (jaylees14) [reply src/main/scala/temple/builder/DockerfileBuilder.scala:29]: Always 😄

**Pull request merged**